### ISSUE TITLE
dzsave with skip_blanks on empty images produces an error

### ISSUE THREAD

**Issue by stastatun** [libvips/pyvips#350]
### Description
When calling `dzsave` with `skip_blanks` on an empty image (i.e. an image in which all pixels are equal to `background`), the function correctly deletes all of the temporary image tiles which correspond to the background. However, this leads to an error later on when moving the temporary folder to `filename_files`, as the directory no longer exists.
### To Reproduce
`libvips 8.12.1`
`pyvips 2-2-1`
```python
import numpy as np
import pyvips

# Non-empty image, saves correctly
image = np.full((1000, 1000, 3), (255, 255, 255), dtype=np.uint8)
image[300:400, 500:600] = (0, 0, 0)
pyvips.Image.new_from_array(image).dzsave(
    "foo",
    skip_blanks=20,
    background=(255, 255, 255),
)

# Empty, crashes
image = np.full((1000, 1000, 3), (255, 255, 255), dtype=np.uint8)
pyvips.Image.new_from_array(image).dzsave(
    "bar",
    skip_blanks=20,
    background=(255, 255, 255),
)
```
```
Traceback (most recent call last):
  File "/home/test.py", line 15, in <module>
    pyvips.Image.new_from_array(image).dzsave(
  File "/usr/local/lib/python3.10/dist-packages/pyvips/vimage.py", line 1347, in call_function
    return pyvips.Operation.call(name, self, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pyvips/voperation.py", line 305, in call
    raise Error('unable to call {0}'.format(operation_name))
pyvips.error.Error: unable to call dzsave
  rename: unable to rename file "./bar-UB0LS1/bar_files" as "./bar_files", No such file or directory
```

**Comment by jcupitt** [comment]
Hi @stastatun,

This was fixed in 8.13. With this prog:

```python
#!/usr/bin/python3

import pyvips

image = (pyvips.Image.black(1000, 1000, bands=3) + 255) \
        .cast("uchar") \
        .copy(interpretation="srgb")

image.dzsave(
    "bar",
    layout="google",
    skip_blanks=20,
    background=(255, 255, 255),
)
```

I now see:

```
john@banana ~/try $ ./skip-blanks.py 
john@banana ~/try $ ls bar/
blank.png  vips-properties.xml
john@banana ~/try $ 
```

**Comment by stastatun** [comment]
Indeed, upgraded to 8.13 and the problem no longer persists.

Thank you for your help! Much appreciated.
